src-cpp/rdkafkacpp.h

```diff
@@ -2927,50 +3031,150 @@ class RD_EXPORT Producer : public virtual Handle {
   };
 
   /**
-   * Transactional API
+   * @name Transactional API
+   * @{
    *
    * Requires Kafka broker version v0.11.0 or later
    *
-   * FIXME: These docs will be updated when the rdkafka.h docs have settled.
+   * See the Transactional API documentation in rdkafka.h for more information.
    */
 
   /**
-   * @brief
+   * @brief Initialize transactions for the producer instance.
+   *
+   * @param timeout_ms The maximum time to block. On timeout the operation
+   *                   may continue in the background, depending on state,
+   *                   and it is okay to call init_transactions() again.
+   *
+   * @returns an RdKafka::Error object on error, or NULL on success.
+   *          Check whether the returned error object permits retrying
+   *          by calling RdKafka::Error::is_retriable(), or whether a fatal
+   *          error has been raised by calling RdKafka::Error::is_fatal().
+   *
+   * @remark The returned error object (if not NULL) must be deleted.
+   *
+   * See rd_kafka_init_transactions() in rdkafka.h for more information.
    *
-   * FIXME blocking?
    */
-  virtual ErrorCode init_transactions (int timeout_ms,
-                                       std::string &errstr) = 0;
+  virtual Error *init_transactions (int timeout_ms) = 0;
+
 
   /**
-   * FIXME blocking?
+   * @brief init_transactions() must have been called successfully
+   *        (once) before this function is called.
+   *
+   * @returns an RdKafka::Error object on error, or NULL on success.
+   *          Check whether a fatal error has been raised by calling
+   *          RdKafka::Error::is_fatal_error().
```

Comment:
Some indication of why a fatal error could occur would be very helpful here i think (pointer to, or brief summary).

Reply:
i felt like i wanted this here, but not in the other methods for some reason. i think because here we're not already in a transaction.

Reply:
Retriable, abortable and fatal errors are explained in rdkafka.h (to which we point from rdkafkacpp.h, to avoid duplication).